### PR TITLE
ChatUI

### LINKED ISSUE #121: Implement ChatUI

### PR DESCRIPTION
**Pull request by jmccook96**
- Set up basics for Chats UI
- Set up a place holder Mock Access object for Chat
- Created FXML and controller
- Edited navigation bar controller to use the final option
- Pre-set up to be prepared for using different chat Ids to differentiate between club chats.
- Implemented on enter sending of message

image

Works on #82 

Resolves #121

Introduces #128 #127 #126

### CODE REVIEW

**Comment by bonnieli0812** [comment]
Looks good but we should look at wrapping text in the messages and input field. Looks like it could be implemented easily enough as part of this PR but also not fussed if you'd prefer to do it in a later one 👍

**Review by bonnieli0812 — APPROVED** [review]

File: target/classes/com/bookclub/chat.fxml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                    </padding>
+                </VBox>
+            </ScrollPane>
+            
+            <!-- Input Field -->
+            <HBox spacing="10" prefHeight="50" alignment="CENTER" style="-fx-background-color: white; -fx-border-color: lightgray; -fx-border-width: 1px 0 0 0;">
+                <padding>
+                    <Insets top="10" right="10" bottom="10" left="10"/>
+                </padding>
+                <TextField fx:id="messageInput" promptText="Type a message..." onAction="#sendMessage" HBox.hgrow="ALWAYS"/>
```
Comment (jmccook96) [inline target/classes/com/bookclub/chat.fxml:33]: #126

File: src/main/java/com/bookclub/controller/ChatController.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    @FXML private VBox       chatBox;
+    @FXML private ScrollPane chatScrollPane;
+    @FXML private TextField  messageInput;
+    @FXML private Label      chatTitleLabel;
+    
+    private IChatAO chatAO;
+    private int currentChatId = 1;
+    
+    @FXML public void initalize() {
+        chatAO = new ChatMAO(); // TODO: Migrate to DAO
```
Comment (jmccook96) [inline src/main/java/com/bookclub/controller/ChatController.java:25]: #127

Comment (jmccook96) [reply src/main/java/com/bookclub/controller/ChatController.java:25]: #128

**Pull request merged**